frontend/src/component/project/Project/Import/ImportModal.tsx

```diff
@@ -49,29 +70,49 @@ export const ImportModal = ({ open, setOpen, project }: IImportModalProps) => {
             <ModalContentContainer>
                 <TimelineContainer>
                     <TimelineHeader>Process</TimelineHeader>
-                    <ImportTimeline stage={importStage.name} />
+                    <ImportTimeline stage={importStage} />
                 </TimelineContainer>
                 <ConditionallyRender
-                    condition={importStage.name === 'configure'}
+                    condition={importStage === 'configure'}
                     show={
                         <ConfigurationStage
-                            project={project}
-                            onClose={() => setOpen(false)}
-                            onSubmit={configuration =>
-                                setImportStage({
-                                    name: 'validate',
-                                    ...configuration,
-                                })
+                            tabs={
+                                <ConfigurationTabs
```

Comment:
this is a technique called "slots not props". Instead of prop drilling data I compose child components where the state is and pass them as ReactNodes.